bin/internal/shared.sh

```diff
@@ -125,12 +122,23 @@ function upgrade_flutter () (
   #  * Contents of STAMP_PATH is not our local git HEAD revision, or
   #  * pubspec.yaml last modified after pubspec.lock
   if [[ ! -f "$SNAPSHOT_PATH" || ! -s "$STAMP_PATH" || "$(cat "$STAMP_PATH")" != "$revision" || "$FLUTTER_TOOLS_DIR/pubspec.yaml" -nt "$FLUTTER_TOOLS_DIR/pubspec.lock" ]]; then
+    # Waits for the update lock to be acquired. Placing this check inside the
+    # conditional allows the majority of flutter/dart installations to bypass
+    # the lock entirely, but as a result this required a second verification that
+    # the SDK is up to date.
+    _wait_for_lock
```

Comment:
Isn't this conceptually the same lock that the windows script acquires in "acquire_lock"? Does that logic need to be updated as well?

Reply:
It does ... but that was more difficult, due to how much longer the check is/gotos